hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -519,6 +498,78 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
                         f"({market_order_record.amount} {market_order_record.base_asset} has been completely filled."
                     )
 
+    cdef bint c_check_if_price_has_drifted(self, object market_pair, LimitOrder active_order):
+        """
+        Given a currently active limit order on maker side, check if its current price is still correct - given the
+        current hedging price on taker market, depth tolerance and transient orders on the maker market captured by
+        recent suggested price samples.
+
+        If the active order's price is no longer the right price, the order will be cancelled.
+
+        This function is only used when active order cancelled is enabled.
+
+        :param market_pair: cross exchange market pair
+        :param active_order: a current active limit order in the market pair
+        :return: True if the order stays, False if the order has been cancelled and we need to re place the orders.
+        """
+        cdef:
+            bint is_buy = active_order.is_buy
+            double order_price = float(active_order.price)
+            double order_quantity = float(active_order.quantity)
+            MarketBase maker_market = market_pair.maker.market
+            OrderBook maker_order_book = market_pair.maker.order_book
+            double suggested_price
+
+        suggested_price = float(self.c_get_market_making_price(market_pair, is_buy, order_quantity))
+
+        top_bid_price, top_ask_price = self.c_get_top_bid_ask_from_price_samples(market_pair)
+
+        if is_buy:
+
+
+            if self._adjust_orders_enabled:
+                adjusting_condition = suggested_price > order_price and suggested_price > top_bid_price
```

Comment:
This only checks for the case where the market moved narrower, but it won't update the price if the market moved wider, to get more potential profit.

I think the `adjust_orders_enabled` flag is not really needed here. But what's needed is that you should do a `suggested_price != order_price` check here instead (make sure they're properly quantized). If they're not equal, then adjust the order.

Reply:
if the market moves the other way, then it will not be profitable and it will get cancelled. its already handled.

Reply:
No. If `suggested_price` is lower than order price, it means you can adjust the order to make more profit.

Reply:
Or put it another way...

The suggested price can become lower because the top of the order book widened, because `c_get_market_making_price()` is using both the effective top of order book price and the taker side hedge price.

Right now, this logic wouldn't adjust the bid pricing in this case, to capture more potential profits.

Reply:
will it not make the bot keep on cancelling and placing orders again and again and make the bot overall unstable ?. I think we should not chase the top order. If someone places above us, then you can chase it.

Reply:
The suggested price is linked to the taker order book as well as the top bid, ask prices. I think making this change, would make it keep on cancelling and replacing the bid and ask prices.

Reply:
That's why we have the anti hysteresis timer. It will only do that every 60 seconds.